src/main/java/budget/Main.java

```diff
@@ -0,0 +1,118 @@
+package budget;
+
+import java.util.Scanner;
+
+public class Main {
+    public static void main(String[] args) {
+        BudgetManager manager = new BudgetManager();
+        Scanner scanner = new Scanner(System.in);
+
+        while (true) {
+            System.out.println("\nHousehold Budget Application");
+            System.out.println("1. Create a new budget");
+            System.out.println("2. Add an expense");
+            System.out.println("3. View saved budgets");
+            System.out.println("4. Edit a budget");
+            System.out.println("5. Edit an expense");
+            System.out.println("6. View budget by date range");
+            System.out.println("7. Delete a budget");
+            System.out.println("8. Exit");
+
+            System.out.print("Enter your choice: ");
+            int choice = scanner.nextInt();
+            scanner.nextLine(); // consume newline
+
+            switch (choice) {
+                case 1:
+                    System.out.println("\nCreate a New Budget:");
+                    System.out.print("Enter the name of the budget: ");
+                    String name = scanner.nextLine();
+                    System.out.print("Enter the total income: ");
+                    double income = 0;
+                    while (true) {
+                        try {
+                            income = Double.parseDouble(scanner.nextLine());
+                            break;
+                        } catch (NumberFormatException e) {
+                            System.out.print("Invalid input. Enter the total income as a number: ");
+                        }
+                    }
+                    System.out.print("Enter the start date (YYYY-MM-DD): ");
+                    String startDate = scanner.nextLine();
+                    System.out.print("Enter the end date (YYYY-MM-DD): ");
+                    String endDate = scanner.nextLine();
+                    manager.createBudget(name, income, startDate, endDate);
+                    break;
+                case 2:
+                    System.out.println("\nAdd an Expense:");
+                    System.out.print("Enter the name of the budget to add an expense: ");
```

Comment:
Usability issue: instead of asking the user to enter the name of the budget each time they want to add an expense, you should give them a list of their budgets to choose from, and then they pick the one they want. They may not remember the exact name they used.

One of the reasons we need a GUI.